web/src/components/core/FileViewer.jsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState, useEffect } from "react";
+import { Popup } from "~/components/core";
+import { Alert, Spinner } from "@patternfly/react-core";
+import cockpit from "../../lib/cockpit";
+
+import "./fileviewer.scss";
+
+// file loading indicator
+const spinner = () => {
```

Comment:
NP: we already have a `Loading` component at https://github.com/yast/d-installer/blob/master/web/src/components/layout/LoadingEnvironment.jsx

Of course, it needs some love starting by the name (we should use something more generic than `LoadingEnvironment`). It could be moved to _components/core_ too.

So, you could use `<LoadingEnvironment title="Reading ${file} file" />`  (although admittedly would be better `<Loader text=...>`)

Anyway, I'm not against for changing the loader icon, but I think we could use only one.